### PR TITLE
chore: Add Shelf Jwt Auth Template

### PR DESCRIPTION
**Pull request by codekeyz**
Closes: [JWT Bearer Auth Template](https://github.com/orgs/invertase/projects/23/views/1?pane=issue&itemId=59829606)

### CODE REVIEW

File: templates/shelf_jwt_auth/lib/router.dart [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  ..post('/register', _register)
+  ..post('/login', _login)
+  ..get('/me', _getCurrentUser);
+
+Middleware authMiddleware = (innerHandler) {
+  return (request) {
+    final reqPath = request.url.path;
+
+    // Bypass auth header check for these routes
+    if (const ['login', 'register'].contains(reqPath)) {
```
Comment (lesnitsky) [inline templates/shelf_jwt_auth/lib/router.dart:22]: is there any other way to check if path matches some route? maybe something is exported from shelf itself?

File: templates/shelf_jwt_auth/lib/router.dart [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      return innerHandler(request);
+    }
+
+    // Retrieve token from header eg: `Bearer xxxxxx` -> `xxxxxx`
```
Comment (lesnitsky) [inline templates/shelf_jwt_auth/lib/router.dart:26]: https://github.com/invertase/globe/pull/73/files#diff-38f5d3106afa19697950a7c86e88f031767c7df053f59ae5f1e64a014ace49d4

File: templates/shelf_jwt_auth/lib/router.dart [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    };
+  } on FormatException catch (e) {
+    return Response.badRequest(body: e.message);
+  }
+
+  if (fauxUserDB.containsKey(reqBody.username)) {
+    return Response.badRequest(body: 'Username already taken.');
+  }
+
+  final String hashedPassword =
```
Comment (lesnitsky) [inline templates/shelf_jwt_auth/lib/router.dart:82]: is type needed here?

File: templates/shelf_jwt_auth/lib/router.dart [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+  return Response.ok('User registered');
+}
+
+Future<Response> _login(Request request) async {
+  ({String username, String password}) reqBody;
+
+  try {
+    final body =
+        await json.decode(await request.readAsString()) as Map<String, dynamic>;
```
Comment (lesnitsky) [inline templates/shelf_jwt_auth/lib/router.dart:99]:
```suggestion
    final bodyString = await request.readAsString();
    final body = json.decode(bodyString) as Map<String, dynamic>;
```

File: templates/shelf_jwt_auth/lib/router.dart [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        ),
+      _ => throw FormatException('Username & Password required for register'),
+    };
+  } on FormatException catch (e) {
+    return Response.badRequest(body: e.message);
+  }
+
+  final userFromDB = fauxUserDB[reqBody.username];
+  if (userFromDB == null ||
+      !BCrypt.checkpw(reqBody.password, userFromDB.passwordHash)) {
```
Comment (lesnitsky) [inline templates/shelf_jwt_auth/lib/router.dart:113]:
```suggestion
  final passwordMatches = BCrypt.checkpw(reqBody.password, userFromDB.passwordHash);

  if (userFromDB == null || !passwordMatches) {
```

**Review by lesnitsky — CHANGES_REQUESTED** [review]

File: templates/shelf_jwt_auth/lib/router.dart (hunk 3 above)
Comment (codekeyz) [reply templates/shelf_jwt_auth/lib/router.dart]: Not really

File: templates/shelf_jwt_auth/lib/router.dart (hunk 1 above)
Comment (codekeyz) [reply templates/shelf_jwt_auth/lib/router.dart:24]: Nope, I haven't found any.

**Review by lesnitsky — APPROVED** [review]

**Pull request merged**